controllers/asosecret_controller_test.go

```diff
@@ -80,7 +81,13 @@ func TestASOSecretReconcile(t *testing.T) {
 						Namespace: "default",
 					}
 				}),
-				getASOAzureClusterIdentity(),
+				getASOAzureClusterIdentity(func(identity *infrav1.AzureClusterIdentity) {
+					identity.Spec.Type = infrav1.IdentityType(defaultClusterClusterIdentityType)
```

Comment:
I wonder if it might be slightly more clear/readable to use the `infrav1.(WorkloadIdentity|ServicePrincipal|...)` constants directly in these tests.

Reply:
The benefit of introducing the variable is that it's clear that `ServicePrincipal` is the default type. I think it'd be fine to change it but maybe you add a comment or something.

Reply:
You're right. This was just a quick fix in order to make lint happy. Adjusted it based on your suggestion.